### ISSUE TITLE
Different downloaded SDL schema files with unchanged remote schema

### ISSUE THREAD

**Issue by francybiga** [apollographql/apollo-tooling#2149]
**Intended outcome:**

When performing the “client:download-schema” multiple times on the same URL (with no changes to the schema) there shouldn’t be changes to the downloaded file. This has been true with version 2.22.1 (and previous versions) but is NOT with version 2.30.2.
I’m downloading the schema using SDL format.

**Actual outcome:**

With version 2.30.2, if I perform the command multiple times, I get a different file (content order is changed) so I have git diffs even if the remote schema isn’t changed

**How to reproduce the issue:**

Using Apollo `2.30.2`
1 - Perform the command `client:download-schema` to download a schema from a remote URL
2 - Commit the downloaded file
3 - Repeat step 1
4 - Although the remote schema hasn't changed, the downloaded file is different from the one committed in step 2

When using a previous version (es `2.22.1`) the downloaded file is the same if the remote schema hasn't be changed.

**Versions**

2.32.2

**Comment by trevor-scheer** [comment]
Hey @francybiga, thanks for reporting this. Can you please provide us with examples of those diffs? Thanks in advance!

**Comment by abernix** [comment]
As an additional point worth clarifying:

You're referring to running the command "multiple times on the same URL" but, but it's worth noting that this command either downloads schemas from the Apollo Registry _OR_ from a running server if you provide an `--endpoint=<URL>` where it can be introspected.

**If you're using `--endpoint`**, what type of GraphQL server is being introspected?  What versions of `graphql` do you have installed in this project (`npm ls graphql`). Have you seen similar indeterministic ordering when browsing the SDL rendered in, e.g., GraphQL Playground's "Schema" view?

**If you're <u>not</u> using `--endpoint`**, that would be good to know!

**Comment by francybiga** [comment]
@trevor-scheer  the differences are in the order of types and operations definitions. 

For example I have a type “Document” defined at line #304. if I download the same schema again, now the type “Document” is defined on line #1968, and so on.

Each time I run the download command the order of the schema changes. In this way git shows me changes even if the content of the schema is the same. This makes spotting what’s changed in the schema (which was the main reason we switched from json to SDL format) very difficult. 
Up to a few months ago (when I opened this issue) the order has always been deterministic, so it was very easy to check any change in the schema.

@abernix we are using the “—endpoint” parameter, specifying our URL. 
When opening the schema in Graphql Playground or Insomia the order is deterministic (it seems in alphabetical order…).
We use `Absinthe 1.4` on the backend, I don't know which version of GraphQL it uses...

**Comment by trevor-scheer** [comment]
@francybiga can you confirm that the introspection query to your running server is stable? This _does_ seem to be a regression across versions of `apollo` if it's simply a matter of changing versions to fix the problem, but I'd like to rule out whatever we can.

I apologize for the frankness, but my focus is currently elsewhere. I'd be happy to look into this if I had a runnable and reliable reproduction of the issue.

**Comment by francybiga** [comment]
Yes, I just tried to perform schema download with `apollo/2.22.1`: the downloaded schema contains types definitions in **alphabetical order**. Instead, when using the latest version `apollo/2.32.1`, the downloaded schema loses the ordering, which changes each time the download is performed.

I managed to workaround this issue by formatting the schema after download using [format-graphql](https://github.com/gajus/format-graphql) but I think it would be useful to restore the ordering in the download command itself as it was before (eventually providing a CLI option).

**Comment by cdavie-weconnect** [comment]
I'm also seeing unstable order from the introspection query with apollo v2.33.2, for what it's worth.
